### DISCUSSION TITLE
add command to update dependencies

### DISCUSSION THREAD

**Discussion by schlichtanders** [rovelstars/reejs#30] [Ideas]
currently there is only `reejs add` but no `reejs update`

**Comment by renhiyama** [comment]
I had ideas about implementing it, but there's a major roadblock. Unlike npm where there's a central rule (api) on where to get newer information about versioning, and other updates, there isn't on esm.sh or any other side.
Also every other URL imports servers: deno.land, esm.sh, jsr.io has their own versioning system, and hence I expected users to update their packages manually.
In my workspace, I managed to make a function where it should be able to take any URL as input, and return back package name & version from it. Maybe I can use it in future to work on this.

**Reply by schlichtanders** [reply]
Thank you for your response. This clarifies a lot.

Might be nice to use it in combination with npm so that import maps simply mimick npm package.json. Then you could also use npm's version information

**Reply by renhiyama** [reply]
I'd try to avoid reading any npm's config/lock files. What I see is npm is good on its own when it comes to installing npm packages, and reejs excels at URL imports. If you think npm is slow, choose other package managers. Reejs current objective is to not break npm, but live together in harmony with npm (and other compatible package managers 😀)
Hence, reejs cannot read npm's lockfiles or anything like that (where npm actually stores useful metadata).
I'll be looking more into integrating both of them soon.
